### PR TITLE
[Ready - 2] FileIO input API

### PR DESCRIPTION
**Pull request by satviksr**
Implements Input API

Documentation: https://github.com/googleinterns/filesystem-in-appsscript/blob/master/docs/FileIO.md

`inputFile` Emulates the VBA `Input` API. The API is designed to be robust to different amount of whitespace and line endings. The API also is designed to support all valid variable types. `inputFileUtil`, `tryInputConstant` and `tryInputVariable` are helper methods for the Input API. They make extensive use of regular expressions and other string manipulations to achieve the required functionality. 

* `inputFileUtil` - Helper function to parse one variable of type `boolean`, `null`, `number`, `string`, `date` or `error`
* `tryInputConstant` - Helper function to parse one constant value such as true, false, null and empty
* `tryInputVariable` - Helper function to parse one variable value - string, numbers, error and 3 forms of date format

### CODE REVIEW

File: FileSystem/FileIOUtil.js [hunk 1]
```diff
@@ -212,3 +212,162 @@ function getLineStart(file) {
...
+    // False Match
+    file.pointer += '#FALSE#'.length;
+    variable.referenceValue = false;
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Helper function to parse variable
```
Comment (monikadhok) [inline FileSystem/FileIOUtil.js]: Please add elaborate comment

File: FileSystem/FileIOUtil.js [hunk 2]
```diff
@@ -212,3 +212,162 @@ function getLineStart(file) {
...
+}
+
+/**
+ * Helper function to parse constant
+ * @param(object) file File structure
+ * @param {VbaBox} variable reference variable to store value
+ * @return {boolean} true if constant is found
+ */
+function tryInputConstant(file, variable) {
+  // Extract next 7 characters only
```
Comment (monikadhok) [inline FileSystem/FileIOUtil.js]: What do you mean by this?

File: FileSystem/FileIOUtil.js [hunk 3]
```diff
@@ -212,3 +212,162 @@ function getLineStart(file) {
   }
   return filePointer + 1;
 }
+
+/**
+ * Helper function to parse one variable
```
Comment (monikadhok) [inline FileSystem/FileIOUtil.js]: Add elaborate comments.

File: FileSystem/FileIOUtil.js [hunk 4]
```diff
@@ -212,3 +212,162 @@ function getLineStart(file) {
...
+    throw new Error('End of file reached');
+  }
+  file.pointer = filePointer;
+  // Try obtaining an input
+  if (!(tryInputConstant(file, variable) || tryInputVariable(file, variable))) {
+    throw new Error('Unknown Expression');
+  }
+  filePointer = file.pointer;
+  // Jump over whitespace and break at any delimiting characters
+  while (filePointer < content.length) {
```
Comment (monikadhok) [inline FileSystem/FileIOUtil.js:256]: Can we club similar conditions, will be more readable.

**Review by monikadhok — CHANGES_REQUESTED** [review]

File: FileSystem/FileIOUtil.js (hunk 4 above)
Comment (satviksr) [reply FileSystem/FileIOUtil.js:256]: Done, reduced 5 cases to 3.

File: FileSystem/FileIOUtil.js (hunk 3 above)
Comment (satviksr) [reply FileSystem/FileIOUtil.js]: Added a bit of documentation

File: FileSystem/FileIOUtil.js (hunk 2 above)
Comment (satviksr) [reply FileSystem/FileIOUtil.js]: Changed the logic slightly, should be clear now.

File: FileSystem/FileIOUtil.js (hunk 1 above)
Comment (satviksr) [reply FileSystem/FileIOUtil.js]: Added a bit of documentation

File: FileSystem/WorkbookTests.js [hunk 5]
```diff
@@ -49,13 +49,6 @@ function workbook_run_all_tests() {
         Workbook.openWorkbook(fileName1);
       });
 
-  QUnit.test(
-      'single call to workbook.open() when file is not present', function() {
-        expect(0);
-        var fileName1 = 'FileNotExist.xlsx';
-        Workbook.openWorkbook(fileName1);
-      });
-
```
Comment (satviksr) [inline FileSystem/WorkbookTests.js:58]: This test is to be replaced with new tests when the complete FileMapper is used.

**Review by monikadhok — DISMISSED** [review]

**Comment by satviksr** [comment]
@monikadhok approved this PR but it is not showing because I resolved merge conflict

**Review by monikadhok — APPROVED** [review]

**Pull request merged**